pkg/agent/wireguard/client_test.go

```diff
@@ -378,3 +379,23 @@ func Test_DeletePeer(t *testing.T) {
 	})
 
 }
+
+func Test_New(t *testing.T) {
+	client := fake.NewSimpleClientset()
+	_, err := New(client, &config.NodeConfig{Name: "test"}, &config.WireGuardConfig{})
+	require.NoError(t, err)
+}
+
+func Test_Init(t *testing.T) {
+	client := fake.NewSimpleClientset()
+	wgClient, _ := New(client,
+		&config.NodeConfig{
+			Name: "test",
+		},
+		&config.WireGuardConfig{
+			Name: "wg-test",
+			MTU:  1450,
+		})
+	err := wgClient.Init()
+	assert.Contains(t, err.Error(), "operation not permitted")
```

Comment:
Is this reliable? Will the output be different if someone runs this test with root or inside a container with `CAP_NET_ADMIN` capability?

Reply:
Good point, should we run it with root capability and check the result? any suggestion?